src/psyclone/tests/psyir/nodes/profile_node_test.py

```diff
@@ -0,0 +1,216 @@
+        profile_node.gen_c_code()
+    assert "Generation of C code is not supported for profiling" \
+        in str(excinfo.value)
+
+
+def test_malformed_profile_node(monkeypatch):
+    ''' Check that we raise the expected error if a ProfileNode does not have
+    a single Schedule node as its child. '''
+    pnode = ProfileNode()
+    monkeypatch.setattr(pnode, "_children", [])
+    with pytest.raises(InternalError) as err:
+        _ = pnode.profile_body
+    assert "malformed or incomplete. It should have a " in str(err.value)
+    monkeypatch.setattr(pnode, "_children", [Node(), Node()])
+    with pytest.raises(InternalError) as err:
+        _ = pnode.profile_body
+    assert "malformed or incomplete. It should have a " in str(err.value)
+
+
+@pytest.mark.parametrize("value", [["a", "b"], ("a"), ("a", "b", "c"),
+                                   ("a", []), ([], "a")])
+def test_profile_node_invalid_name(value):
+    '''Test that the expected exception is raised when an invalid profile
+    name is provided to a ProfileNode.
+
+    '''
+    with pytest.raises(InternalError) as excinfo:
+        _ = ProfileNode(options={"region_name": value})
+    assert ("Error in PSyDataNode. The name must be a tuple containing "
+            "two non-empty strings." in str(excinfo.value))
+
+
+def test_lower_to_lang_level_single_node(parser):
+    ''' Test the lower_to_language_level() method when a Schedule contains
+    a single ProfileNode.
+
+    '''
+    Profiler.set_options([Profiler.INVOKES])
+    symbol_table = SymbolTable()
+    arg1 = symbol_table.new_symbol(
+        symbol_type=DataSymbol, datatype=REAL_TYPE)
+    zero = Literal("0.0", REAL_TYPE)
+    one = Literal("1.0", REAL_TYPE)
+    assign1 = Assignment.create(Reference(arg1), zero)
+    assign2 = Assignment.create(Reference(arg1), one)
+
+    kschedule = KernelSchedule.create(
+        "work1", symbol_table, [assign1, assign2, Return()])
+    Profiler.add_profile_nodes(kschedule, Loop)
+    assert isinstance(kschedule.children[0], ProfileNode)
+    assert isinstance(kschedule.children[-1], Return)
+    kschedule.lower_to_language_level()
+    # The ProfileNode should have been replaced by two CodeBlocks with its
+    # children inserted between them.
+    assert isinstance(kschedule[0], CodeBlock)
+    # The first CodeBlock should have the "profile-start" annotation.
+    assert kschedule[0].annotations == ["profile-start"]
+    ptree = kschedule[0].get_ast_nodes
+    assert len(ptree) == 1
+    assert isinstance(ptree[0], Fortran2003.Call_Stmt)
+    assert kschedule[1] is assign1
+    assert kschedule[2] is assign2
+    assert isinstance(kschedule[-2], CodeBlock)
+    assert kschedule[-2].annotations == []
+    ptree = kschedule[-2].get_ast_nodes
+    assert len(ptree) == 1
+    assert isinstance(ptree[0], Fortran2003.Call_Stmt)
+    assert isinstance(kschedule[-1], Return)
+    # Check that the symbol table contains the appropriate symbols:
```

Comment:
Are the lines from here to the end of the test something that the constructor is currently in charge of . Would it make more sense to check this is the constructor test*?

*or somewhere else, see previous comment

Reply:
I've moved them into the constructor test and added a TODO.